components/layout_2020/dom_traversal.rs

```diff
@@ -236,8 +252,23 @@ fn traverse_element<'dom, Node>(
             }
         },
         Display::GeneratingBox(display) => {
-            let contents =
-                replaced.map_or(NonReplacedContents::OfElement.into(), Contents::Replaced);
+            let is_text_control_element: bool = matches!(
+                element.type_id(),
+                LayoutNodeType::Element(LayoutElementType::HTMLInputElement)
+            ) || matches!(
+                element.type_id(),
+                LayoutNodeType::Element(LayoutElementType::HTMLTextAreaElement)
+            );
```

Comment:
This is repeated from above, so maybe add a `fn is_text_control_element` helper.

Otherwise, `: bool` is unnecessary and you can merge `matches!()`.